docs/mvvm/ObservableRecipient.md

```diff
@@ -31,18 +31,18 @@ public class MyViewModel : ObservableRecipient, IRecipient<LoggedInUserRequestMe
 }
 ```
 
-In the above, `OnActivated` automatically registers the instance as a recipient for `LoggedInUserRequestMessage` messages, using that method as the action to invoke. Using the `IRecipient<TMessage>` interface is not mandatory, and the registration can also be done manually (even using just an inline lambda expression):
+In the example above, `OnActivated` automatically registers the instance as a recipient for `LoggedInUserRequestMessage` messages, using that method as the action to invoke. Using the `IRecipient<TMessage>` interface is not mandatory, and the registration can also be done manually (even using just an inline lambda expression):
 
 ```csharp
 public class MyViewModel : ObservableRecipient
 {
     protected override void OnActivated()
     {
         // Using a method group...
-        Messenger.Register<LoggedInUserRequestMessage>(this, Receive);
+        Messenger.Register<MyViewModel, LoggedInUserRequestMessage>(this, (r, m) => r.Receive(m));
```

Comment:
We needed the parent type for significant performance, eh? No way around that? 😉 We'd probably need source generators or something too heavy/fancy, right?

Reply:
@michael-hawker I included an extension method that offers the non-generic version too 😊
Basically you can do either of these:

```csharp
Messenger.Register<LoggedInUserRequestMessage>(this, (r, m) => ((MyViewModel)r).Receive(m));
// or
Messenger.Register<MyViewModel, LoggedInUserRequestMessage>(this, (r, m) => r.Receive(m));
```

The issue with the former is that the input recipient is the just an `object`, so you need that cast in the lambda expression. That's the reason why I came up with that weird hack to enable generic support here, so that you get build-time safety for the type checks and never need to cast your recipients.